compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -9790,6 +9918,129 @@ TR::Node *lnegSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    return node;
    }
 
+TR::Node *fpNegSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
+   {
+   TR::Node *firstChild;
+
+   if (TR::Compiler->arith.permitsFusedMultiplyAdd() && s->cg()->supportsNegativeFusedMultiplyAdd())
+      {
+      firstChild = node->getFirstChild();
+      if (firstChild->getOpCode().isMulAdd() &&
+          performTransformation(s->comp(), "%sTransforming [" POINTER_PRINTF_FORMAT "] neg of fma\n", s->optDetailString(), node))
+         {
+         switch(firstChild->getOpCodeValue())
+            {
+            case TR::fmuladd:
+               TR::Node::recreate(firstChild, TR::fnegmulsub);
+               break;
+            case TR::fmulsub:
+               TR::Node::recreate(firstChild, TR::fnegmuladd);
+               break;
+            case TR::fnegmuladd:
+               TR::Node::recreate(firstChild, TR::fmulsub);
+               break;
+            case TR::fnegmulsub:
+               TR::Node::recreate(firstChild, TR::fmuladd);
+               break;
+            case TR::dmuladd:
+               TR::Node::recreate(firstChild, TR::dnegmulsub);
+               break;
+            case TR::dmulsub:
+               TR::Node::recreate(firstChild, TR::dnegmuladd);
+               break;
+            case TR::dnegmuladd:
+               TR::Node::recreate(firstChild, TR::dmulsub);
+               break;
+            case TR::dnegmulsub:
+               TR::Node::recreate(firstChild, TR::dmuladd);
+               break;
+            default:
+                return node;
```

Comment:
I would prefer adding the 3, I think, missing cases (the vector opcodes) to return node, then an assert for anything else.